src/app/shared/pricing-definitions/pricing-definitions-table-data-source.ts

```diff
@@ -102,32 +112,28 @@ export class PricingDefinitionsTableDataSource extends DialogTableDataSource<Pri
   }
 
   public buildTableColumnDefs(): TableColumnDef[] {
-    const tableActions: TableColumnDef[] = [
+    return [
       {
         id: 'name',
         name: 'chargers.name',
         headerClass: 'col-15p',
         class: 'col-15p',
         sortable: true,
-      }
-    ];
-    if (this.viewingAllComponents) {
-      tableActions.push(...[
-        {
-          id: 'entityType',
-          name: 'transactions.dialog.session.pricing_detail_entity_type',
-          headerClass: 'col-15p',
-          class: 'col-15p',
-        },
-        {
-          id: 'entityName',
-          name: 'transactions.dialog.session.pricing_detail_entity_name',
-          headerClass: 'col-15p',
-          class: 'col-15p',
-        }
-      ]);
-    }
-    tableActions.push(
+      },
+      {
+        id: 'entityType',
+        name: 'transactions.dialog.session.pricing_detail_entity_type',
+        headerClass: 'col-15p',
+        class: 'col-15p',
+        visible: this.viewingAllComponents
+      },
+      {
+        id: 'entityName',
+        name: 'transactions.dialog.session.pricing_detail_entity_name',
+        headerClass: 'col-15p',
+        class: 'col-15p',
+        visible: this.viewingAllComponents
+      },
```

Comment:
@ClaudeROSSI, the "new way" was not working as expected, it always showed a column that was in the projectField array, even if in the table column definition we set visible = false.
I added a fix so that we can control the visibility of a column despite it being in the projected field array.